### PR TITLE
Feature 112

### LINKED ISSUE #112: Entering an incorrect password when changing from citizen to guardian logs out the user
When you click the _Change to Guardian_ button in the giraf app bar, you're prompted for a password.
If this password does not match the password of the user, you are instead logged out.

Expected behavior is:
- On correct password: The mode is changed to guardian mode.
- On incorrect password: The dialog box informs the user that the password is incorrect, without signing them out.

Originally reported in: #109

### PR DESCRIPTION
**Pull request by User12837985123**
This closes #112.

There is not given a picture since the issue did now change anything in the UI.

### CODE REVIEW

**Comment by User12837985123** [comment]
@aau-giraf/sw610f19  you have been assigned this. Here's your checklist. I marked @Moesgaarda  as the reviewer, but you might change it as you like.
Please use the following checklist. Feel free to check it off as you go along.
**Code Design**
- [ ] Is the code in the right place? (Both in terms of folder structure and class structure)
- [ ] Could this code, to the best of your knowledge, have reused existing code?
- [ ] Is the code over-engineered? Examples of over-engineering: Implemented behavior for future problems
- [ ] Does it introduce functionality that is not necessary for solving the problem?

**Code Readability and Maintainability**
- [ ] Are names meaningful and self-documenting?
- [ ] Is it understandable, by reading the code, what it does?
- [ ] Is the code simple and readable? i.e. it contains no “hack”, or obscure solution.
- [ ] Is it understandable, by reading the tests, what they do?
- [ ] Do the tests cover sensible cases? Both happy paths and exception paths?
- [ ] Do the tests cover the full functionality? i.e., will the tests fail if some of the requested functionality is missing?
- [ ] Are error messages in the code understandable?
- [ ] Are confusing sections of code covered by comments?

**Functionality**
For this part, open the app, and test:
- [ ] Are the features that the code claims to implement, actually implemented?
- [ ] Does it look like the code has bugs?
- [ ] Are all new screens reachable through in-app navigation?

**Points for Discussion and Reflection**
- [ ] Are UI messages, that the user might meet, actually helpful?
- [ ] Are there errors that are bad practice? Like, hard-coding a test database connection?
- [ ] Is there well written code in this? Remember to tell the developer!

**Comment by User12837985123** [comment]
@aau-giraf/sw613f19  you have been assigned this. Here's your checklist. I marked @ralled96  as the reviewer, but you might change it as you like.
Please use the following checklist. Feel free to check it off as you go along.
**Code Design**
- [ ] Is the code in the right place? (Both in terms of folder structure and class structure)
- [ ] Could this code, to the best of your knowledge, have reused existing code?
- [ ] Is the code over-engineered? Examples of over-engineering: Implemented behavior for future problems
- [ ] Does it introduce functionality that is not necessary for solving the problem?

**Code Readability and Maintainability**
- [ ] Are names meaningful and self-documenting?
- [ ] Is it understandable, by reading the code, what it does?
- [ ] Is the code simple and readable? i.e. it contains no “hack”, or obscure solution.
- [ ] Is it understandable, by reading the tests, what they do?
- [ ] Do the tests cover sensible cases? Both happy paths and exception paths?
- [ ] Do the tests cover the full functionality? i.e., will the tests fail if some of the requested functionality is missing?
- [ ] Are error messages in the code understandable?
- [ ] Are confusing sections of code covered by comments?

**Functionality**
For this part, open the app, and test:
- [ ] Are the features that the code claims to implement, actually implemented?
- [ ] Does it look like the code has bugs?
- [ ] Are all new screens reachable through in-app navigation?

**Points for Discussion and Reflection**
- [ ] Are UI messages, that the user might meet, actually helpful?
- [ ] Are there errors that are bad practice? Like, hard-coding a test database connection?
- [ ] Is there well written code in this? Remember to tell the developer!

**Comment by Alechj** [comment]
Hello!
We tried out your functionality and we have 2 questions/comments.
In the issue it says: "On incorrect password: The dialog box informs the user that the password is incorrect, without signing them out". However it does not seem like the user is warned when he types a wrong password, is this intended?

The second comment is that if you type in a password and then click the close button on the dialog, the password is still there the next time you open the dialog which might not be intended.

We will start reviewing as soon as these 2 things are resolved.

File: lib/widgets/giraf_app_bar_widget.dart [hunk 1]
```diff
@@ -52,53 +52,55 @@ class GirafAppBar extends StatelessWidget implements PreferredSizeWidget {
...
-                          ],
-                        ),
+                key: const Key('changeUserMode'),
+                icon: Image.asset('assets/icons/changeToCitizen.png'),
+                tooltip: 'Skift til borger tilstand',
+                onPressed: () {
+                  Alert(
+                      context: context,
+                      style: _alertStyle,
+                      title: 'Skift til værge',
```
Comment (Tekrus) [inline lib/widgets/giraf_app_bar_widget.dart:62]: Billedet siger change to citizen, tooltip siger skift til borgertilstand...

Tror her skal stå borger 🙂

**Review by Tekrus — CHANGES_REQUESTED** [review]

**Pull request closed without merging**